parser/bootstrap/lark_parser.py

```diff
@@ -0,0 +1,101 @@
+import json
+from pathlib import Path
+from typing import Any
+
+from lark import Lark, Token, Tree, UnexpectedInput
+
+from codegen.user_facing_errors import InvalidSyntax
+
+GRAMMAR_PATH = Path(__file__).parent / "grammar.lark"
+lark = Lark.open(
+    str(GRAMMAR_PATH),
+    parser="earley",
+    ambiguity="resolve",
+    start="program",
+    propagate_positions=True,
+)
+
+
+def token_to_dict(token: Token) -> dict[str, Any]:
+    return {
+        "name": token.type,
+        "value": token.value,
+        "meta": {
+            "start": {"line": token.line, "column": token.column},
+            "end": {"line": token.end_line, "column": token.end_column},
+        },
+    }
+
+
+def tree_to_dict(tree: Tree) -> dict[str, Any]:
```

Comment:
This might be cleaner with a `json.JSONEncoder`

Reply:
I looked into this a bit... Looks like there's no way to do it more concisely than this without a more strongly typed interface. I plan to make this better once we've removed lark tho.